libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -784,12 +838,18 @@ public static ExpressionEvaluator TimeTransform(string type, Func<DateTime, int,
                     string error = null;
                     IReadOnlyList<object> args;
                     (args, error) = EvaluateChildren(expr, state, options);
+                    var locale = DetermineLocaleInfo(state);
```

Comment:
> DetermineLocaleInfo [](start = 32, length = 20)

This is fine as a default, but I should be able to explicitly pass in the locale in all of these places.

Reply:
I see. I change the order of evaluating the locale to ensure that error messages are arranged in the right order. If a locale string was passed explicitly, then it won't try to access the locale stored in the memory.
If no locale given, and locale stored in memory is illegal, it will throw an error.
If no locale given and no locale info stored in memory, take the default locale.